src/layers/basic.jl

```diff
@@ -692,3 +692,126 @@ end
+  a vector of class indices and `offset` should be a vector representing the starting index of a bag in the `inputs` vector. The first element of `offsets` must be `1`, and `offsets` should
+  be monotonically increasing, but the second condition is not checked.
+
+  This format is useful for dealing with flattened representations of "ragged" tensors. E.g., if you have a flat vector of class labels that need to be grouped in a non-uniform way. However, under the hood, it is just syntactic sugar for the vector-of-vectors input style.
+
+  For example, the `input`/`offset` pair `[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]`/`[1, 5, 6, 8]`
+  is equivalent to the bags `[[1, 2, 3, 4], [5], [6, 7], [8, 9, 10]]`, since the first bag starts at index `1` and goes up to index `5`, non-inclusive. The next bag starts at index `5` and goes up to index `6`, non-inclusive, etc. Below is another example usage.
+
+# Examples
+```jldoctest
+julia> vocab_size, embed_size = 1000, 4;
+
+julia> model = Flux.EmbeddingBag(vocab_size => embed_size)
+EmbeddingBag(1000 => 4)  # 4_000 parameters
+
+julia> bags = [[1, 200, 25, 789], [2, 5, 10, 999]];
+
+julia> bags_mtx = [1 2; 200 5; 25 10; 789 999];
+
+julia> model(bags) |> summary
+"4×2 Matrix{Float32}"
+
+julia> model(bags) ≈ model(bags_mtx)
+true
+```
+
+```jldoctest
+julia> vocab_size, embed_size = 10, 8;
+
+julia> model = Flux.EmbeddingBag(vocab_size => embed_size)
+EmbeddingBag(10 => 8)  # 80 parameters
+
+julia> scalar_bag = 5 # just a single bag of one item
+5
+
+julia> model(scalar_bag);
+
+julia> single_bag = [1, 2, 2, 4]; # one bag several items
+
+julia> model(single_bag) |> summary
+"8-element Vector{Float32}"
+
+julia> bags_mtx = [1 2 3; 4 5 6] # 2 bags each with 3 items
+2×3 Matrix{Int64}:
+ 1  2  3
+ 4  5  6
+
+julia> model(bags_mtx) |> summary
+"8×3 Matrix{Float32}"
+
+julia> vec_vec_bags = [[1, 2], [3], [4], [5, 6, 7]]; # 4 bags with different number of items.
+
+julia> model(vec_vec_bags) |> summary
+"8×4 Matrix{Float32}"
+
+julia> inputs = [1, 4, 5, 2, 3];
+
+julia> offsets = [1, 3, 4]; # 3 bags of sizes [2, 1, 2]
+
+julia> model(inputs, offsets) |> summary
+"8×3 Matrix{Float32}"
+
+julia> oh_bag = Flux.OneHotVector(2, vocab_size); # single bag of one item
+
+julia> model(oh_bag) |> summary
+"8-element Vector{Float32}"
+
+julia> ohm_bag = Flux.OneHotMatrix([2, 3, 5, 7], vocab_size); # 4 bags, each with one item
+
+julia> model(ohm_bag) |> summary
+"8×4 Matrix{Float32}"
+```
+"""
+struct EmbeddingBag{F, W}
+  weight::W
+  reduction::F
+end
+
+@functor EmbeddingBag
+
+EmbeddingBag((in, out)::Pair{<:Integer, <:Integer}, reduction::Function = mean; init = randn32) = EmbeddingBag(init(out, in), reduction)
+EmbeddingBag(weight) = EmbeddingBag(weight, mean)
+
+function (m::EmbeddingBag)(inputs::AbstractVector, offsets::AbstractVector)
+    offsets[firstindex(offsets)] == 1 || throw(ArgumentError("`offsets` must begin with 1."))
+    start = firstindex(inputs)
+    newoffsets = vcat(offsets, [lastindex(inputs)])
```

Comment:
This only accepts `AbstractVector{<:Integer}`, and doesn't seem to use `start`.
```suggestion
function (m::EmbeddingBag)(inputs::AbstractVector{<:Integer}, offsets::AbstractVector{<:Integer})
    first(offsets) == 1 || throw(ArgumentError("`offsets` must begin with 1."))
    newoffsets = vcat(offsets, lastindex(inputs))
```
It seems to be allowing for OffsetArrays, but nothing else in Flux will work with these. Can it be simplified much if you put `LinearAlgebra.require_one_based_indexing(inputs, offsets)` somewhere?

Reply:
I removed `start` which was unused. I don't think it can be simplified much, with that check, as `first`/`lastindex` etc will do basically what we would otherwise do by hand.